features/messages/impl/src/test/kotlin/io/element/android/features/messages/impl/pinned/list/PinnedMessagesListPresenterTest.kt

```diff
@@ -189,9 +190,20 @@ class PinnedMessagesListPresenterTest {
             skipItems(3)
             val filledState = awaitItem() as PinnedMessagesListState.Filled
             val eventItem = filledState.timelineItems.first() as TimelineItem.Event
+
+            pinnedEventsTimeline.unpinEventLambda = successUnpinEventLambda
+            filledState.eventSink(PinnedMessagesListEvents.HandleAction(TimelineItemAction.Unpin, eventItem))
+
+            pinnedEventsTimeline.unpinEventLambda = failureUnpinEventLambda
             filledState.eventSink(PinnedMessagesListEvents.HandleAction(TimelineItemAction.Unpin, eventItem))
+
             cancelAndIgnoreRemainingEvents()
-            assert(unpinEventLambda)
+
+            assert(successUnpinEventLambda)
```

Comment:
The assertion should be done earlier, else the test does not check the order of the lambda invocation.